src/controllers/restful/PaymentController.ts

```diff
@@ -307,3 +320,56 @@ export function getActivationStatus(
 function generateCodiceContestoPagamento(): CodiceContestoPagamento {
   return uuid.v1().replace(/-/g, "") as CodiceContestoPagamento;
 }
+
+/**
+ * Parse a PagoPa response to check if it contains functional errors.
+ * If error is found, it is mapped into a controller response for BackendApp
+ * @param {string} esito - The esito (OK or KO) provided by PagoPa
+ * @param {string} faultBean - Optional information provided by PagoPa in case of error
+ * @return {IResponseErrorGeneric | IResponseErrorInternal} A controller response or undefined if no errors exist
+ */
+export function getResponseErrorIfExists(
+  esito: string,
+  faultBean: faultBean_ppt
+): IResponseErrorGeneric | IResponseErrorInternal {
+  // Case 1: Response is SUCCESS
+  if (esito === "OK") {
+    return undefined;
+  }
+  // Case 2: Response is FAILED but no additional information are provided by PagoPa
+  if (faultBean === undefined) {
+    return ResponseErrorInternal("Error during payment check: esito === KO");
+  }
+  // Case 3: Response is FAILED and additional information are provided by PagoPa
+  const errorMessageCtrl = getErrorMessageCtrlFromPagoPaError(
+    faultBean.faultCode
+  );
+  return ResponseErrorGeneric(
+    HttpStatusCodeEnum.HTTP_STATUS_403,
+    errorMessageCtrl,
+    faultBean.faultCode,
+    faultBean.faultString
+  );
+}
+
+/**
+ * Convert PagoPa message error (faultCode) to Controller message error (ErrorMessagesCtrlEnum) to send to BackendApp
```

Comment:
would you mind providing a link to the relevant pagopa docs that describe the errors in the function comment?